library/Icinga/Common/DataExtractor.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Icinga\Common;
+
+trait DataExtractor
```

Comment:
I'm not sure about the justification of this trait. Method `toArray` is used once and not implemented anyway, so can be dropped. `fromArray` is named like a factory method but isn't one (doesn't return a new object). The trait itself is also only used once. I'd like to propose that you remove this trait and add the functionality of `fromArray` as part of the method `BaseDashboard::setProperties` and use that instead.

Though, please only where the code really doesn't know which properties need to be set. There are some usages in the method `loadDashboardEntries` of various classes, which should exactly know which setters to call, so they should. Helps also to know for sure that those setters are called and where if inspected using the IDE.